src/main/java/core/basesyntax/dao/impl/CommentDaoImpl.java

```diff
@@ -3,30 +3,79 @@
 import core.basesyntax.dao.CommentDao;
 import core.basesyntax.model.Comment;
 import java.util.List;
+import lombok.extern.log4j.Log4j;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
 import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
 
+@Log4j
 public class CommentDaoImpl extends AbstractDao implements CommentDao {
     public CommentDaoImpl(SessionFactory sessionFactory) {
         super(sessionFactory);
     }
 
     @Override
     public Comment create(Comment entity) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(entity);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                log.debug("Transaction creation for " + entity.toString()
+                        + " has been rollbacked.", e);
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert Content entity");
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        log.debug("Entity " + entity.toString() + " created");
+        return entity;
     }
 
     @Override
     public Comment get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return session.get(Comment.class, id);
+        }
     }
 
     @Override
     public List<Comment> getAll() {
-        return null;
+        try (Session session = factory.openSession()) {
+            Query<Comment> getAllCommentQuery = session.createQuery("from Comment", Comment.class);
+            return getAllCommentQuery.getResultList();
+        }
     }
 
     @Override
     public void remove(Comment entity) {
-
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            Comment comment = session.find(Comment.class, entity.getId());
```

Comment:
do we need to get an object before removing?

Reply:
It is two ways to remove the object from db if to do it like this, you are avoiding Exception, if remove without check you will get Exception when this object do not exist. So it means you have to do an additional check, is id that is giving into the method is correct.  I'll simplify it.